plugins/utils/better-escape.nix

```
@@ -0,0 +1,73 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.better-escape;
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options.plugins.better-escape =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "better-escape.nvim";
+
+      package = helpers.mkPackageOption "better-escape.nvim" pkgs.vimPlugins.better-escape-nvim;
+
+      mapping = helpers.mkNullOrOption (with types; listOf str) ''
+        List of mappings to use to enter escape mode.
+      '';
+
+      timeout =
+        helpers.defaultNullOpts.mkNullable
+        (with types;
+            either ints.unsigned str)
+        "vim.o.timeoutlen" ''
+          The time in which the keys must be hit in ms. Use option timeoutlen by default.
+        '';
+
+      clearEmptyLines = helpers.defaultNullOpts.mkBool false ''
+        Clear line after escaping if there is only whitespace.
+      '';
+
+      keys =
+        helpers.defaultNullOpts.mkNullable
+        (with types;
+            either str helpers.rawType)
+        "<ESC>"
+        ''
+          Keys used for escaping, if it is a function will use the result everytime.
+
+          example(recommended):
+
+          keys = {
+          	_raw = \'\'
+          		function()
+          			return vim.api.nvim_win_get_cursor(0)[2] > 1 and '<esc>l' or '<esc>'
+          		 end,.
+          		 \'\'
+          	};
+        '';
+    };
```

No, it cannot be a list.
The option declaration is fine.

Minor formatting changes:
```suggestion
      keys =
        helpers.defaultNullOpts.mkNullable
        (with types;
          either str helpers.rawType
        )
        "<ESC>"
        ''
          Keys used for escaping, if it is a function will use the result everytime.

          Example (recommended):

          keys.__raw = \'\'
            function()
              return vim.api.nvim_win_get_cursor(0)[2] > 1 and '<esc>l' or '<esc>'
            end
          \'\';
        '';
    };
```